src/Tes/Models/TesTaskBackendParametersUtilityMethods.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace Tes.Models
+{
+    public partial class TesTask
```

Comment:
I suggest moving these methods to existing class TesTaskExtensions and making the methods extensions for TesResources, for example: 
`public static string GetBackendParameterValue(this TesResources tesResources, TesResources.SupportedBackendParameters parameter)`
I think this is a bit cleaner as these new parameters are "resources", not top level task properties. SO the usage will be